### PR TITLE
Masthead: add demo that includes horizontal nav

### PR DESCRIPTION
**Pull request by ajaypratap003**
**What**: Closes https://github.com/patternfly/patternfly-react/issues/6826
Masthead: add demo that includes horizontal nav

**Additional issues**:

### CODE REVIEW

**Review by wise-king-sullyman — COMMENTED** [review]
Looks great!

One question I have, I wonder if we would want to remove all of the logic for the search input and the favoriting and everything? It seems like it isn't really the focus of this demo and it seems to take up about half of the file.

File: packages/react-core/src/demos/Masthead.md [hunk 1]
```diff
@@ -24,3 +24,7 @@ Many mastheads contain utilities which collapse into a dropdown at smaller viewp
 ```ts file='./examples/Masthead/MastheadWithUtilitiesAndUserDropdownMenu.tsx' isFullscreen
 
 ```
+### Masthead with horizontal nav
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/Masthead.md:27]: Can we just update this and the above demo name to omit the "Masthead" text?

File: packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx [hunk 2]
```diff
@@ -0,0 +1,561 @@
...
+    </Nav>
+  );
+
+  const masthead = (
+    <Masthead>
+      <MastheadToggle>
+        <PageToggleButton variant="plain" aria-label="Global navigation">
+          <BarsIcon />
+        </PageToggleButton>
+      </MastheadToggle>
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx:507]: We don't need this MastheadToggle

File: packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx [hunk 3]
```diff
@@ -0,0 +1,561 @@
...
+
+    return () => {
+      window.removeEventListener('keydown', handleMenuKeys);
+      window.removeEventListener('click', handleClickOutside);
+    };
+  }, [isOpen, menuRef]);
+
+  const toggle = (
+    <MenuToggle
+      aria-label="Toggle"
```
Comment (thatblindgeye) [inline packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx:152]:
```suggestion
      aria-label="Applications"
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

**Comment by tlabaj** [comment]
@ajaypratap003 this demo was recently added to the v6 branch.  Perhaps you could cherry pick that commit, or you could make sure they align.

**Comment by ajaypratap003** [comment]
> @ajaypratap003 this demo was recently added to the v6 branch. Perhaps you could cherry pick that commit, or you could make sure they align.

@tlabaj Can you please share the commit which is related to this PR

**Comment by tlabaj** [comment]
> > @ajaypratap003 this demo was recently added to the v6 branch. Perhaps you could cherry pick that commit, or you could make sure they align.
> 
> @tlabaj Can you please share the commit which is related to this PR

@ajaypratap003 Nevermind.  I was mistaken.  it was a Nav example that was added.

**Review by thatblindgeye — APPROVED** [review]

**Comment by ajaypratap003** [comment]
@tlabaj Can you please merge this PR if everything look good.

**Review by mattnolting — CHANGES_REQUESTED** [review]
Nice work! Looks like you need to add `toolbar` to `.pf-v5-c-masthead__content`.

Screenshot 2024-04-30 at 1 32 47 PM

**Comment by ajaypratap003** [comment]
> Nice work! Looks like you need to add `toolbar` to `.pf-v5-c-masthead__content`.
> 
> Screenshot 2024-04-30 at 1 32 47 PM

@mattnolting Sorry, I don't get this point. Can you please describe bit more

**Review by kmcfaul — APPROVED** [review]
lgtm

@mattnolting Can you explain a bit more what your requested change is? The masthead content does contain the toolbar.

**Comment by mattnolting** [comment]
@kmcfaul @ajaypratap003 
From: https://www.patternfly.org/components/masthead/html-demos/horizontal-nav/

Screenshot 2024-05-08 at 12 44 27 PM

**Review by mattnolting — CHANGES_REQUESTED** [review]
`nav` goes inside a `toolbar__item` with the class `pf-m-overflow-container`

**Current**

Screenshot 2024-05-08 at 2 20 31 PM



**update**
Screenshot 2024-05-08 at 2 20 19 PM

**Comment by ajaypratap003** [comment]
> `nav` goes inside a `toolbar__item` with the class `pf-m-overflow-container`
> 
> **Current**
> 
> Screenshot 2024-05-08 at 2 20 31 PM
> **update** Screenshot 2024-05-08 at 2 20 19 PM

nav is already inside toolbar_item. this is the latest commit https://github.com/patternfly/patternfly-react/pull/10241/commits/525f4a7b7b89695cc40fc1a4d7fac8d79aa6ce00

**Comment by mattnolting** [comment]
> > `nav` goes inside a `toolbar__item` with the class `pf-m-overflow-container`
> > **Current**
> > Screenshot 2024-05-08 at 2 20 31 PM
> > **update** Screenshot 2024-05-08 at 2 20 19 PM
> 
> nav is already inside toolbar_item. this is the latest commit [525f4a7](https://github.com/patternfly/patternfly-react/commit/525f4a7b7b89695cc40fc1a4d7fac8d79aa6ce00)

Still can't see these. 
```
This private-user-images.githubusercontent.com page can’t be found
```

File: packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+              </NavItem>
+              <NavItem itemId={5} isActive={activeItem === 4} to="#horizontal-6">
+                Horizontal nav item 6
+              </NavItem>
+              <NavItem itemId={6} isActive={activeItem === 4} to="#horizontal-7">
+                Horizontal nav item 7
+              </NavItem>
+            </NavList>
+          </Nav>
+        </ToolbarItem>
```
Comment (mattnolting) [inline packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx:272]: As only a single toolbar should be used, this needs to be moved to line 234

File: packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx [hunk 5]
```diff
@@ -0,0 +1,330 @@
...
+                isExpanded={isDropdownOpen}
+                icon={<Avatar src={imgAvatar} alt="" />}
+              >
+                Ned Username
+              </MenuToggle>
+            )}
+          >
+            <DropdownList>{userDropdownItems}</DropdownList>
+          </Dropdown>
+        </ToolbarItem>
```
Comment (mattnolting) [inline packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx:233]:
```suggestion
        </ToolbarItem>
        <PageHorizontalNav />

```

File: packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+      </ToolbarContent>
+    </Toolbar>
+  );
+
+  const PageHorizontalNav = () => (
+    <Toolbar id="toolbar-nav" isFullHeight>
+      <ToolbarContent>
```
Comment (mattnolting) [inline packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx:240]:
```suggestion
```

File: packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+                Horizontal nav item 6
+              </NavItem>
+              <NavItem itemId={6} isActive={activeItem === 4} to="#horizontal-7">
+                Horizontal nav item 7
+              </NavItem>
+            </NavList>
+          </Nav>
+        </ToolbarItem>
+      </ToolbarContent>
+    </Toolbar>
```
Comment (mattnolting) [inline packages/react-core/src/demos/examples/Masthead/MastheadWithHorizontalNav.tsx:274]:
```suggestion
```

**Review by mattnolting — COMMENTED** [review]
Almost there, need to include nav in a single toolbar

**Review by mattnolting — APPROVED** [review]
Great work! Looks like you got it all 👍

**Pull request merged**